seisbench/generate/labeling.py

```diff
@@ -454,6 +454,132 @@ def __str__(self):
             f"ProbabilisticPointLabeller (label_type={self.label_type}, dim={self.dim})"
         )
 
+class PolarityPlusLabeller(PickLabeller):
+    r"""
+    - Creates probabilistic supervised labels from seismic polarity information, extending the PickLabeller base class
+    - Supports both single-window (2D) and multi-window (3D) input data
+    - Labels include three classes: Up, Down, and Noise
+
+    Label Generation:
+    - Generates probability distributions at specified onset positions based on polarity information ('U' for Up, 'D' for Down)
+    - Noise class is automatically calculated as the remaining probability: noise = max(0, 1 - (up + down))
+    - Supports three probability distribution shapes: Gaussian, triangle, and box
+
+    Shape Parameters:
+    - Gaussian: sigma represents the standard deviation (in samples)
+    - Triangle: sigma represents the half-width (in samples)
+    - Box: sigma represents the half-width (in samples)
+
+    Input/Output Dimensions:
+    - 2D input (samples×channels): outputs (3×samples)
+    - 3D input (windows×samples×channels): outputs (windows×3×samples)
+
+    Note: The parameter sigma has different interpretations depending on the chosen shape. Picks with NaN sample values are treated as not present.
+    """
+
+    def __init__(self, polarity_column: str = "trace_polarity", shape: str = "gaussian", sigma: int = 10, **kwargs):
+        self.polarity_column = polarity_column
+        self.label_method = "probabilistic"
+        self.sigma = sigma
+        self.shape = shape
+        self._labelshape_fn_mapper = {
+            "gaussian": gaussian_pick,
+            "triangle": triangle_pick,
+            "box": box_pick,
+        }
+        kwargs["dim"] = kwargs.get("dim", 1)
+        super().__init__(label_type="multi_class", **kwargs)
+
+    def label(self, X, metadata):
```

Comment:
Given this function is in large part a duplicate of `ProbabilisticLabeller.label`, I think it would be good to unify the two. From what I can see, the only thing differing is what exactly is written into the y value. Maybe the inheritance could be something like:
`PickLabeller`
-> `StreamPickLabeller` (would host most of the function with an if-statement switching between the labels`
----> `ProbabilisticLabeller` & `PolarityPlusLabeller` (essentially just setting a configuration parameter to ensure the function in the super class does the right thing.